src/server/scripts/Commands/cs_cast.cpp

```diff
@@ -191,13 +202,7 @@ class cast_commandscript : public CommandScript
         if (!*args)
             return false;
 
-        Unit* target = handler->getSelectedUnit();
-        if (!target)
-        {
-            handler->SendSysMessage(LANG_SELECT_CHAR_OR_CREATURE);
-            handler->SetSentErrorMessage(true);
-            return false;
-        }
+        Unit* player = handler->GetSession()->GetPlayer();
```

Comment:
that is not a good idea, you should be able to select another creature and let him cast the spell